analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -68,20 +69,33 @@ private void ProcessObjectCreation(ProgramState state, IObjectCreationOperationW
         }
     }
 
-    private static ProgramState ProcessInvocation(ProgramState state, IInvocationOperationWrapper invocation) =>
-        IsCryptographicallyStrongRandomNumberGenerator(invocation)
-        && invocation.ArgumentValue("data") is { } dataArgument
-        && dataArgument.TrackedSymbol() is { } trackedSymbol
-            ? state.SetSymbolConstraint(trackedSymbol, ByteCollectionConstraint.CryptographicallyStrong)
-            : state;
+    private static ProgramState ProcessInvocation(ProgramState state, IInvocationOperationWrapper invocation)
+    {
+        if (IsCryptographicallyStrongRandomNumberGenerator(invocation)
+            && invocation.Arguments.FirstOrDefault() is { } dataArgument
```

Comment:
I wouldn't just take the first or default - I would be more precise - ask the first or default that is also a byte array.

In theory these methods should have only one argument right?